app/services/fact_parser.rb

```diff
@@ -152,6 +152,19 @@ def find_interface_by_name(host_name)
           nil
         end
       end
+
+      if (ip = values[:ipaddress6]).present?
+        begin
+          logger.debug { "Resolving fact '#{ip}' via DNS to match reported hostname #{host_name}" }
+          if resolver.getnames(ip).any? { |name| name.to_s == host_name }
+            logger.debug { "Match: '#{ip}', interface #{int} is selected as primary" }
+            return [int, values]
```

Comment:
I know this was already the case before, but this can also be
```suggestion
            return true
```